File: problem_003/problem_003.rb
def prime_factors(number)
  factors = []
  test_number = 2
  while number >= test_number
    if number % test_number == 0
      number /= test_number
      factors << test_number
    else
      test_number += 1
    end
  end
  return factors
end

puts prime_factors(600851475143).max

# Since we started at the number 2, any factors we get will have to be prime.     

File: problem_003/problem_003.py
def max_prime_factor(number):
	first_count = 2
	factors = []
	while number >= first_count:
		if number % first_count == 0:
			factors.append(first_count)
			number /= first_count
		else:
			first_count += 1
	return max(factors)

print max_prime_factor(600851475143 )